### PR TITLE
Local car refactoring

### PR DESCRIPTION
**Pull request by LucasBrazi06**
Changed car catalog json format to match the eMobility one
Aligned Tata image namings
Resized images
Check hash to not import if the car catalog is not changed
Introduced Json schema to import car catalog

### CODE REVIEW

**Comment by LucasBrazi06** [comment]
@mahdibenromdhane I refactored and improved car catalog local import.
Could you check and test it?
Could you also check the Job that sync car catalogs with ev-database?
Thanks.
I would like to merge this PR today.

File: src/storage/mongodb/CarStorage.ts [hunk 1]
```diff
@@ -143,164 +144,16 @@ export default class CarStorage {
...
-      euroNCAPSA: Utils.convertToInt(carToSave.euroNCAPSA),
-      relatedVehicleIDSuccessor: Utils.convertToInt(carToSave.relatedVehicleIDSuccessor),
-      eVDBDetailURL: carToSave.eVDBDetailURL,
-      videos: carToSave.videos,
-      hash: carToSave.hash,
-      imageURLs: carToSave.imageURLs,
-      image: carToSave.image,
-      imagesHash: carToSave.imagesHash
+      _id: carToSave.id,
+      ...carCatalog,
```
Comment (Ramzay) [inline src/storage/mongodb/CarStorage.ts:153]: ',' after carCatalog should be removed

File: src/storage/mongodb/validator/CarValidatorStorage.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  private carCatalog: Schema;
+
+  private constructor() {
+    super('CarValidatorStorage', {
+      strict: false, // When 'true', it fails with anyOf required fields: https://github.com/ajv-validator/ajv/issues/1571
+      allErrors: true,
+      removeAdditional: false, // 'all' fails with anyOf documents: Manually added 'additionalProperties: false' in schema due filtering of data in anyOf/oneOf/allOf array (it's standard): https://github.com/ajv-validator/ajv/issues/1784
+      allowUnionTypes: true,
+      coerceTypes: true,
+      verbose: true,
```
Comment (Ramzay) [inline src/storage/mongodb/validator/CarValidatorStorage.ts:18]: Last ',' should be removed

File: src/storage/mongodb/CarStorage.ts (hunk 1 above)
Comment (LucasBrazi06) [reply src/storage/mongodb/CarStorage.ts:153]: Thanks for your feedback, we usually let the ending comma to ease the cut'n paste in a Json document :-)

**Review by mahdibenromdhane — APPROVED** [review]

**Comment by mahdibenromdhane** [comment]
> @mahdibenromdhane I refactored and improved car catalog local import. Could you check and test it? Could you also check the Job that sync car catalogs with ev-database? Thanks. I would like to merge this PR today.

Done, I'v pushed a small fix : the image thumb was not saved in LocalCarCatalogBootstrap.

**Comment by mahdibenromdhane** [comment]
> > 
> 
> > > @mahdibenromdhane I refactored and improved car catalog local import. Could you check and test it? Could you also check the Job that sync car catalogs with ev-database? Thanks. I would like to merge this PR today.
> > 
> > 
> > Done, I'v pushed a small fix : the image thumb was not saved in LocalCarCatalogBootstrap.
> 
> Thanks Mahdi! Did you check the car catalog synchronization also?

Yes :)

**Pull request merged**